data/env/snapd.fish.in

```diff
@@ -8,7 +8,14 @@ end
 # looked for in XDG_DATA_DIRS; make sure it includes the relevant directory for
 # snappy applications' desktop files.
 set -ul snap_xdg_path /var/lib/snapd/desktop
-set --path --export XDG_DATA_DIRS $XDG_DATA_DIRS
-if not contains $snap_xdg_path $XDG_DATA_DIRS
-    set XDG_DATA_DIRS $XDG_DATA_DIRS $snap_xdg_path
+# note, snapd supports distros going back as far as Ubuntu 16.04, what means we
+# must make sure that this is compatible with old fish shells which do not have
+# fish_add_path or set --path
+set --export XDG_DATA_DIRS $XDG_DATA_DIRS
```

Comment:
In fish, "exported" is not a scope. You probably want to explicitly request a global variable:

```fish
set --global --export XDG_DATA_DIRS $XDG_DATA_DIRS
```

Reply:
(this isn't necessary for $PATH because that's guaranteed to be global)